data/actions/scripts/other/teleport.lua

```diff
@@ -1,5 +1,10 @@
 local upFloorIds = {1386, 3678, 5543}
 function onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	local tile = item:getTile()
+	if tile and tile:hasFlag(TILESTATE_PROTECTIONZONE) and player:isPzLocked() then
```

Comment:
> You should also or actually only check destination tile.
> There should also be a check on the rope script side to prevent players from roping others that are PZ locked to PZ zone.
> 
> Anyway, I don't think this should be addressed at all, since this is not the way the map should be designed.

Well, I'm not sure how this works on Tibia. I don't think there are rope spots on protected areas and if there are, both the hole and the spot have a wide protection zone.